src/op_analytics/cli/subcommands/pulls/defillama.py

```diff
@@ -88,60 +84,102 @@ def process_breakdown_stables(data):
         "price",
     ]
 
-    metadata = {}
-    for key in must_have_metadata_fields:
-        metadata[key] = data[key]
-
-    for key in metadata_fields:
-        metadata[key] = data.get(key)
-
-    result = pl.DataFrame(rows, infer_schema_length=len(rows)).with_columns(
-        id=pl.lit(metadata["id"]),
-        name=pl.lit(metadata["name"]),
-        symbol=pl.lit(metadata["symbol"]),
+    metadata: Dict[str, Optional[str]] = {
+        key: data.get(key) for key in must_have_metadata_fields + metadata_fields
+    }
+
+    result = (
+        pl.DataFrame(rows, infer_schema_length=len(rows)).with_columns(
+            id=pl.lit(metadata["id"]),
+            name=pl.lit(metadata["name"]),
+            symbol=pl.lit(metadata["symbol"]),
+        )
+        if rows
+        else pl.DataFrame()
     )
 
     return result, metadata
 
 
-def pull_stables():
-    """Pull stablecoin data from DeFiLlama.
+def pull_stables(
+    stablecoin_ids: Optional[List[str]] = None, days: int = 30
+) -> Dict[str, pl.DataFrame]:
+    """
+    Pulls and processes stablecoin data from DeFiLlama.
 
-    - Fetch the stablecoins summary endpoint.
-    - For each stablecoin, fetch the detailed breakdown data.
-    - Write all results to BigQuery.
+    Args:
+        stablecoin_ids: List of stablecoin IDs to process. Defaults to None (process all).
+        days: Number of days of data to retrieve. Defaults to 30.
+
+    Returns:
+        A dictionary containing metadata and breakdown DataFrames.
     """
     session = new_session()
     summary = get_data(session, SUMMARY_ENDPOINT)
 
-    urls = {}
-    for stablecoin in summary["peggedAssets"]:
-        stablecoin_id = stablecoin["id"]
-        urls[stablecoin_id] = BREAKDOWN_ENDPOINT.format(id=stablecoin_id)
+    all_stablecoin_ids = [asset["id"] for asset in summary.get("peggedAssets", [])]
```

Comment:
I think we must not have a fallback for `peggedAssets` like this:
```
summary.get("peggedAssets", [])
```
We want the script to fail if the `peggedAssets` key is not present.